packages/amplify-codegen-e2e-tests/schemas/admin-modelgen.graphql

```diff
@@ -0,0 +1,50 @@
+input AMPLIFY { globalAuthRule: AuthRule = { allow: public } } # FOR TESTING ONLY!
+
+type Todo @model {
+  id: ID!
+  name: String!
+  description: String
+  phone: Phone
+}
+type Phone {
+  number: String
+}
+enum BillingSource {
+  CLIENT
+  PROJECT
+}
+input CustomInput {
+  customField1: String!
+  customField2: BillingSource
+  customField3: NestedInput!
+}
+input NestedInput {
+  content: String! = "hello"
+}
+interface ICustom {
+  firstName: String!
+  lastName: String
+  birthdays: [INestedCustom!]!
+}
+interface INestedCustom {
+  birthDay: AWSDate!
+}
+# The member types of a Union type must all be Object base types.
+union CustomUnion = Todo | Phone
+
+type Query {
+  getAllTodo(msg: String, input: CustomInput): String @function(name: "echofunction-${env}")
+  echo(msg: String): String
+  echo2(todoId: ID!): Todo
+  echo3: [Todo!]!
+  echo4(number: String): Phone
+  echo5: [CustomUnion!]!
+  echo6(customInput: CustomInput): String!
+  echo7: [ICustom]!
```

Comment:
The type of the custom operation inputs can also be list of scalar, enum or input object types. Can we add those cases as well in this test? Would suggest making some of those required as well so we cover that case too? [Spec reference](https://spec.graphql.org/October2021/#sec-Types)